### PR TITLE
DOP-4758: pull rstspec from tagged version 

### PR DESCRIPTION
**Pull request by seungpark**
### Stories/Links:

DOP-4758

### Notes
- [Prereq PR](https://github.com/mongodb/docs-worker-pool/pull/1058) for makefile changes
- This PR aims to get rid of `latest` references to rstspec and points these to the latest tag

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

File: src/commands/src/shared/next-gen-parse.ts [hunk 1]
```diff
@@ -3,7 +3,10 @@ import { Job } from '../../../entities/job';
 import { getDirectory } from '../../../job/jobHandler';
 import { CliCommandResponse, ExecuteCommandError, executeCliCommand } from '../helpers';
 
-const RSTSPEC_FLAG = '--rstspec=https://raw.githubusercontent.com/mongodb/snooty-parser/latest/snooty/rstspec.toml';
+const SNOOTY_PARSER_VERSION = process.env['SNOOTY_PARSER_VERSION']
+  ? `v${process.env['SNOOTY_PARSER_VERSION']}`
+  : 'main';
+const RSTSPEC_FLAG = `--rstspec=https://raw.githubusercontent.com/mongodb/snooty-parser/${SNOOTY_PARSER_VERSION}/snooty/rstspec.toml`;
```
Comment (mmeigs) [inline src/commands/src/shared/next-gen-parse.ts:9]: I forget, does this need a pre-pended `v` for the version tag?

Comment (rayangler) [inline src/commands/src/shared/next-gen-parse.ts:9]: Do we need the `RSTSPEC_FLAG` at all? I believe by default, the parser should just use whatever version of the rstspec it has

Comment (rayangler) [reply src/commands/src/shared/next-gen-parse.ts:9]: Sorry, redundant comment. Just saw [the comment](https://github.com/mongodb/docs-worker-pool/pull/1058#issuecomment-2189754742) from the other PR.

Comment (seungpark) [reply src/commands/src/shared/next-gen-parse.ts]: yup! removed it according to the same reason. and removed the ARG -> ENV changes as we won't need the SNOOTY_PARSER_VERSION specified

Comment (seungpark) [reply src/commands/src/shared/next-gen-parse.ts]: see comment below! removed the flag

**Review by mmeigs — APPROVED** [review]

**Pull request merged**